crates/ruff_python_ast/src/source_code/indexer.rs

```diff
@@ -59,25 +64,34 @@ impl Indexer {
 
             match tok {
                 Tok::Comment(..) => {
-                    commented_lines.push(*range);
+                    comment_ranges.push(*range);
                 }
                 Tok::Newline | Tok::NonLogicalNewline => {
                     line_start = range.end();
                 }
                 Tok::String {
-                    triple_quoted: true,
+                    triple_quoted,
+                    kind,
                     ..
-                } => string_ranges.push(*range),
+                } => {
+                    if *triple_quoted {
+                        triple_quoted_string_ranges.push(*range);
+                    }
+                    if matches!(kind, StringKind::FString | StringKind::RawFString) {
+                        f_string_ranges.push(*range);
```

Comment:
We now track f-string ranges, which lets us do a binary search to find the containing f-string whenever we need to fix within an f-string.